rust/operator-binary/src/druid_controller.rs

```diff
@@ -150,6 +157,31 @@ pub async fn reconcile_druid(
             cm_name: zk_confmap.clone(),
         })?;
 
+    // Assemble the OPA connection string from the discovery and the given path, if a spec is given.
+    let opa_connstr = if let Some(opa_spec) = &druid.spec.opa {
+        let opa_confmap = opa_spec.opa_config_map_name.clone();
+        let opa_host = client
+            .get::<ConfigMap>(&opa_confmap, druid.namespace().as_deref())
+            .await
+            .context(GetOpaConnStringConfigMapSnafu {
+                cm_name: opa_confmap.clone(),
+            })?
+            .data
+            .and_then(|mut data| data.remove("OPA"))
+            .context(MissingOpaConnStringSnafu {
+                cm_name: zk_confmap.clone(),
+            })?;
+
+        // Add the API path to the end
+        Some(
+            opa_host.trim_end_matches('/').to_owned()
+                + "/"
+                + opa_spec.opa_druid_api_path.trim_start_matches('/'),
+        )
+    } else {
+        None
+    };
```

Comment:
We usally mount the env var now from the configmaps and work with the envar from there on (https://github.com/stackabletech/kafka-operator/blob/26a074abf60f9243540ac34076a97c713a89cba3/rust/operator/src/kafka_controller.rs#L520).